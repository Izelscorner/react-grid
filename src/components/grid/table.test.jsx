import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Table from './table';

describe('<Table />', () => {
    configure({adapter: new Adapter()});
    let wrapper;
    let data = [{name: 'test', lastname: 'something'}, {name: 'test', lastname: 'something'}];
    let columns = [{columnHeader: 'Name', columnField: 'name'},{columnHeader: 'Lastname', columnField: 'lastname'}]
    
    beforeAll(() => {
        wrapper = shallow(<Table data={data} columns={columns} isLoading={false} />);
      });

    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Header Text', () => {
        const theadColumns = wrapper.find('table thead tr th');
        expect(theadColumns.first().text()).toEqual('Name');
        expect(theadColumns.last().text()).toEqual('Lastname');
      });
  
    it('renders 2 rows', () => {
      const rows = wrapper.find('table tbody tr');
      expect(rows).toHaveLength(2);
    });
  
    it('render 2 columns', function () {
      const columns = wrapper.find('table tbody tr').first().find('td');
      expect(columns).toHaveLength(2);
    });

    it('render is loading', function () {
        wrapper = shallow(<Table data={data} columns={columns} isLoading={true}   />);
        const row = wrapper.find('table tbody tr td');
        expect(row.text()).toEqual('Loding...');
      });
});
