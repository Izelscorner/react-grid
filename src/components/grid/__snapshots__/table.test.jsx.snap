// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> matches snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        key="0"
      >
        Name
      </th>
      <th
        key="1"
      >
        Lastname
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        className="font-weight-bold"
        key="0"
      >
        test
      </td>
      <td
        className=""
        key="1"
      >
        something
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        className="font-weight-bold"
        key="0"
      >
        test
      </td>
      <td
        className=""
        key="1"
      >
        something
      </td>
    </tr>
  </tbody>
</table>
`;
